crypto/rand/rand_uniform.c

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2023 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include "crypto/rand.h"
+#include "internal/common.h"
+
+/*
+ * Implementation an optimal random integer in a range function.
+ * Refer: https://github.com/apple/swift/pull/39143 for a description
+ * of the algorithm.
+ */
```

Comment:
I'd prefer a complete description of the algorithm here in the source code instead just a link to a pull request in another library. I doesn't have to be as detailed as the original comments, but contain the main essence.

Reply:
Okay, can do but not today.

Reply:
Outline of algorithm added.